src/tribol/mesh/MeshData.cpp

```diff
@@ -694,75 +588,75 @@ void MeshData::computeNodalNormals( int const dim )
       allocIntArray( &numFaceNrmlsToNodes, size, 0 );
    } // end if-check on null mesh
 
-   // loop over cells
-   for (int i=0; i<this->m_numCells; ++i)
+   // loop over elements
+   for (int i=0; i<this->numberOfElements(); ++i)
    {
-      // loop over cell nodes
-      for (int j=0; j<this->m_numNodesPerCell; ++j)
+      // loop over element nodes
+      for (int j=0; j<this->numberOfNodesPerElement(); ++j)
       {
 
          // SRW: note the connectivity array must be local to the mesh for indexing into 
          // the mesh nodal normal array. If it is not, then nodeId will access some other
          // piece of memory and there may be a memory issue when numFaceNrmlsToNodes is deleted
          // at the end of this routine.
-         integer nodeId = this->m_connectivity[ this->m_numNodesPerCell * i + j ]; 
-         m_node_nX[ nodeId ] += this->m_nX[ i ]; // m_nX[i] is the ith face normal x-component
-         m_node_nY[ nodeId ] += this->m_nY[ i ]; // see above...
+         int nodeId = getGlobalNodeId(i, j);
+         m_node_n(0, nodeId) += this->m_n[0][ i ]; // m_n[0][i] is the ith face normal x-component
+         m_node_n(1, nodeId) += this->m_n[1][ i ]; // see above...
 
          // increment face-normal-to-node contribution counter
          ++numFaceNrmlsToNodes[ nodeId ];
 
-      } // end loop over cell nodes
+      } // end loop over element nodes
 
       // populate z-component for 3D problems
       if (dim == 3)
       {
 
-         // repeat loop over cell nodes for z-component
-         for (int k=0; k<this->m_numNodesPerCell; ++k)
+         // repeat loop over element nodes for z-component
+         for (int k=0; k<this->numberOfNodesPerElement(); ++k)
          {
-            integer nodeId = this->m_connectivity[ this->m_numNodesPerCell * i + k ]; 
-            m_node_nZ[ nodeId ] += this->m_nZ[ i ];
-         } // end loop over cell nodes
+            int nodeId = getGlobalNodeId(i, k); 
+            m_node_n[2][ nodeId ] += this->m_n[2][ i ];
```

Comment:
`m_node_n[2][ nodeId ]` is different compared to above where you have e.g. `m_node_n(1, nodeId)`. Can you clarify? Would it make sense to throw this calculation inside a loop over dimension since we are now storing the data this way?

Reply:
I didn't bother since it will get refactored soon, but I went ahead and made the change.